api/src/main/java/org/openmrs/module/queue/api/QueueEntryService.java

```diff
@@ -113,4 +123,13 @@ String generateVisitQueueNumber(@NotNull Location location, @NotNull Queue queue
 	 * @param sortWeightGenerator the SortWeightGenerator to set
 	 */
 	void setSortWeightGenerator(SortWeightGenerator sortWeightGenerator);
+	
+	/**
+	 * Given a specified queue entry, return its previous queue entry (i.e. the queue entry the patient
+	 * transition from to get to the specified one)
```

Comment:
It isn't really an "i.e." condition.  There are specific rules and conditions here that we should document - specifically that the queue entries must have matching `queue`, `patient`, and `visit`, and the `endedAt` of the returned QueueEntry must match the `startedAt` of the passed QueueEntry exactly.